compiler/java/src/main/java/com/c/lexer/Lexer.java

```diff
@@ -0,0 +1,29 @@
+package com.c.lexer;
+
+public class Lexer {
+
+    // --- Comment chars
+    private final String commentLineIdentifier;
+    private final String commentBlockIdentifierOpen;
+    private final String commentBlockIdentifierClose;
+
+    private enum Context {
+        PRORGRAM_CODE, // Strings before or after a conditional section
```

Comment:
Would not call that `PROGRAM_CODE` because in e.g. yaml or json files it is not program code but data. In the C++ implementation I called it `ARBITRARY`.

Reply:
Yeah, I saw that but `ARBITRARY` looked to unspecific to me. But now I know the reason behind it, I forgot about data file formats... Will call it arbitrary as well :D